src/main/java/seedu/module/model/module/UniqueModuleList.java

```diff
@@ -77,6 +77,27 @@ public void setModules(List<TrackedModule> trackedModules) {
         internalList.setAll(trackedModules);
     }
 
+    /**
+     * Replaces the module {@code target} in the list with {@code editedModule}.
+     * {@code target} must exist in the list.
+     * The module identity of {@code editedModule} must not be the same as another existing module in the list.
+     */
+    public void setModule(TrackedModule target, TrackedModule editedPerson) {
+        requireAllNonNull(target, editedPerson);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new ModuleNotFoundException();
+        }
+
+        if (!target.isSameModule(editedPerson) && contains(editedPerson)) {
+            throw new DuplicateModuleException();
+        }
```

Comment:
This code is very weird. So if the target module and the editedModule is not the same (via identity), they're a duplicate? Can we remove this branch?